reproduce-figures/prepare-scpca-portal-data.R

```diff
@@ -0,0 +1,309 @@
+expected_projects <- readLines(opts$project_whitelist)
+
+# Define and check input projects
+input_zips <- list.files(
+  opts$portal_projects_dir, 
+  pattern = "^SCPCP\\d{6}_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_\\d+{4}-\\d+{2}-\\d+{2}\\.zip",
+  full.names = TRUE
+) |>
+  # name by project
+  purrr::set_names(
+    \(x) {stringr::str_split_i(basename(x), "_", 1)}
+  )
+stopifnot(
+  "The provided input directory does not contain all expected projects. Zip files for all projects listed in the `project-whitelist.txt` file should be present." = 
+    setequal(names(input_zips), expected_projects)
+)
+
+
+# Copy over the merged SCPCP000003 file ----------------
+output_merged_sce_path <- file.path(output_dir_s3_files, "SCPCP000003", "SCPCP000003_merged.rds")
+fs::dir_create(dirname(output_merged_sce_path))
+fs::file_copy(
+  opts$merged_sce_path, 
+  output_merged_sce_path
+)
+
+
+# Map over project zip files to organize files for reproducibility ------------------
+input_zips |>
+  purrr::iwalk(
+    \(project_zip, project_id){
+      
+      # temporary variables used for testing
+      #project_id <- "SCPCP000001"
+      #project_zip <- "/Users/sjspielman/Downloads/projects/SCPCP000001_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_2025-04-03.zip"
+      
+      project_consensus_dir <- file.path(output_dir_consensus_files, project_id)
+      
+      # Unzip the directory, specifying the target in consensus
+      unzip(project_zip, exdir = project_consensus_dir)
+
+      # First, remove all the qc htmls
+      system(glue::glue("rm {project_consensus_dir}/*/*html"))
+      
+      # Define all sample directories present in this project, _excluding_ multiplexed samples
+      # This is because multiplexed samples aren't used in any figures created from ScPCA data files
+      sample_dirs <- list.dirs(
+        project_consensus_dir, 
+        full.names = TRUE
+      ) |>
+        purrr::set_names(\(x) {basename(x)}) |>
+        # get rid of the project folder & multiplexed samples
+        purrr::discard_at(\(x) {stringr::str_detect(x, "SCPCP\\d{6}")}) |>
+        purrr::discard_at(\(x) {stringr::str_detect(x, "_")}) 
+
+      ####### Step 1: Copy RDS files to target directories #######
+      sample_dirs |>
+        purrr::iwalk(
+          \(sample_dir, sample_id) {
+            
+            # Copy processed rds files to celltype_results_files
+            # The next step will parse these files into TSVs
+            if (sample_id %in% celltype_results_samples) {
+              system(glue::glue("cp {sample_dir}/*processed.rds {output_dir_celltype_results}"))
+            } 
+            
+            # Copy rds files to standalone sample dir
+            if (sample_id %in% standalone_sample_dirs) {
+              system(glue::glue("cp -r {sample_dir} {output_dir_s3_files}"))
+            } 
+          
+            # We can now remove the (un)filtered files
+            system(glue::glue("rm {sample_dir}/*filtered.rds"))
```

Comment:
I think if we are going to remove files then it probably makes sense to just remove the unzipped folder after doing all the copying rather than removing individual files?